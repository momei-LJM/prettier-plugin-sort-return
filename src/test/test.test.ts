import { expect, it } from "vitest";
import { tryFrmatFile } from ".";
import { resolve } from "path";
const reslovePath = (path: string) => {
  return resolve(process.cwd(), path);
};
it("test format vue sfc", async () => {
  const result = await tryFrmatFile(reslovePath("src/examples/vue/index.vue"));
  expect(result).toMatchSnapshot();
});

it("test format vue sfc-setup", async () => {
  const result = await tryFrmatFile(
    reslovePath("src/examples/vue/setupVue.vue")
  );
  expect(result).toMatchSnapshot();
});

it("test format typescript", async () => {
  const result = await tryFrmatFile(reslovePath("src/examples/ts/index.ts"));
  expect(result).toMatchSnapshot();
});

it("test format javascript", async () => {
  const result = await tryFrmatFile(reslovePath("src/examples/js/index.js"));
  expect(result).toMatchSnapshot();
});
