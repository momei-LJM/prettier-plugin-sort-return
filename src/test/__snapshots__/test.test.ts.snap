// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`test format javascript 1`] = `
"export default defineComponent({
  setup() {
    const NOT_NULL_CONST = "NOT_NULL_CONST";
    const NOT_NULL_CONST2 = "NOT_NULL_CONST2";
    const SELECT_NOT_NULL_CONST = "SELECT_NOT_NULL_CONST";

    const a = 1;
    const foo = 2;
    const bar = 3;
    const baz = 4;
    return {
      a,
      foo,
      baz,
      bar,
      NOT_NULL_CONST,
      NOT_NULL_CONST2,
      SELECT_NOT_NULL_CONST,
    };
  },
});
"
`;

exports[`test format typescript 1`] = `
"export default defineComponent({
  setup() {
    const NOT_NULL_CONST = "NOT_NULL_CONST";
    const NOT_NULL_CONST2 = "NOT_NULL_CONST2";
    const SELECT_NOT_NULL_CONST = "SELECT_NOT_NULL_CONST";

    const a = 1;
    const foo = 2;
    const bar = 3;
    const baz = 4;
    return {
      a,
      foo,
      baz,
      bar,
      NOT_NULL_CONST,
      NOT_NULL_CONST2,
      SELECT_NOT_NULL_CONST,
    };
  },
});
"
`;

exports[`test format vue sfc 1`] = `
"<template>
  <div class="flex">
    <el-button type="primary"> submit </el-button>
  </div>
</template>

<script lang="ts">
import { defineComponent } from "vue";
export default defineComponent({
  setup() {
    const NOT_NULL_CONST = "NOT_NULL_CONST";
    const NOT_NULL_CONST2 = "NOT_NULL_CONST2";
    const SELECT_NOT_NULL_CONST = "SELECT_NOT_NULL_CONST";

    const a = 1;
    const foo = 2;
    const bar = 3;
    const baz = 4;
    return {
      a,
      foo,
      baz,
      bar,
      NOT_NULL_CONST,
      NOT_NULL_CONST2,
      SELECT_NOT_NULL_CONST,
    };
  },
});
</script>

<style lang="scss" scoped>
.flex {
  display: flex;
}
</style>
"
`;

exports[`test format vue sfc-setup 1`] = `
"<template>
  <div class="flex">
    <el-button type="primary"> submit </el-button>
  </div>
</template>

<script lang="ts" setup>
const fn = () => {
  const NOT_NULL_CONST = "NOT_NULL_CONST";
  const NOT_NULL_CONST2 = "NOT_NULL_CONST2";
  const SELECT_NOT_NULL_CONST = "SELECT_NOT_NULL_CONST";

  const a = 1;
  const foo = 2;
  const bar = 3;
  const baz = 4;
  return {
    a,
    foo,
    baz,
    bar,
    NOT_NULL_CONST,
    NOT_NULL_CONST2,
    SELECT_NOT_NULL_CONST,
  };
};
</script>

<style lang="scss" scoped>
.flex {
  display: flex;
}
</style>
"
`;
